### PR TITLE
User-saveable circuits

### PR DESCRIPTION
**Pull request by KevinMackenzie**
Adds a back-end and (TODO) front-end for storing user circuits on the same server as web server.  Initially, this supports SQLite, but should be flexible enough to support other database types and storage strategies.

- [x] Make sure the metadata updates all the right fields on update

### CODE REVIEW

File: site/go/api/circuits.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	storageInterface.UpdateCircuit(circuit)
+
+	// Returned the updated metadata so the client can get any changes the server made to it
+	c.XML(http.StatusAccepted, circuit.Metadata)
+}
+
+func circuitLoadHandler(m interfaces.CircuitStorageInterfaceFactory, c *gin.Context, userId model.UserId) {
+	circuitId, err := strconv.ParseInt(c.Param("id"), 10, 64)
+	if err != nil {
+		c.XML(http.StatusBadRequest, err.Error())
```
Comment (LeonMontealegre) [inline site/go/api/circuits.go]: should there be a return statement after this?
@KevinMackenzie

Comment (KevinMackenzie) [reply site/go/api/circuits.go]: yes

**Comment by KevinMackenzie** [comment]
I'm getting errors in `Circuits.ts`, `Ping.ts`, and `Example.ts` complaining that jQuery `$.when...` isn't compatible with `Promise`s.  Is there a new installation step, or are you also having this issue?

**Comment by LeonMontealegre** [comment]
> I'm getting errors in `Circuits.ts`, `Ping.ts`, and `Example.ts` complaining that jQuery `$.when...` isn't compatible with `Promise`s. Is there a new installation step, or are you also having this issue?

Try to run `npm install` again?

**Comment by KevinMackenzie** [comment]
it looks like I had to `npm update`.  `npm install` alone didn't fix it.

**Pull request merged**